examples/tools/generate.ipynb

```diff
@@ -0,0 +1,502 @@
+    "we want to have a higher density of phase-space points near the resonance.\n",
+    "\n",
+    "In case of a narrow resonance, **importance sampling** is beneficial. Instead\n",
+    "of creating an unweighted uniform phase space sample, we generate a weighted\n",
+    "sample with more points in the peak regions."
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "We first again create an amplitude model, this time including a 'widened'\n",
+    ":math:`\\omega` resonance (see also :doc:`particles`):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pycompwa.expertsystem.ui.system_control import StateTransitionManager\n",
+    "from pycompwa.expertsystem.ui.system_control import InteractionTypes\n",
+    "from pycompwa.expertsystem.amplitude.helicitydecay import \\\n",
+    "    HelicityAmplitudeGeneratorXML\n",
+    "from pycompwa.expertsystem.state.particle import particle_list\n",
+    "\n",
+    "initial_state = [(\"J/psi\", [-1, 1])]\n",
+    "final_state = [(\"gamma\"), (\"pi0\"), (\"pi0\")]\n",
+    "tbd_manager = StateTransitionManager(initial_state, final_state,\n",
+    "                                     formalism_type='helicity',\n",
+    "                                     topology_building='isobar')\n",
+    "\n",
+    "# Widen omega resonance\n",
+    "omega = particle_list[\"omega(782)\"]\n",
+    "parameters = omega['DecayInfo']['Parameter']\n",
+    "for par in parameters:\n",
+    "    if par['@Type'] == 'Width':\n",
+    "        par['Value'] = 0.001\n",
+    "\n",
+    "tbd_manager.set_allowed_interaction_types(\n",
+    "    [InteractionTypes.Strong, InteractionTypes.EM])\n",
+    "tbd_manager.allowed_intermediate_particles = ['f2(1270)', 'omega']\n",
+    "graph_interaction_settings_groups = tbd_manager.prepare_graphs()\n",
+    "solutions, _ = tbd_manager.find_solutions(graph_interaction_settings_groups)\n",
+    "\n",
+    "print(\"found \" + str(len(solutions)) + \" solutions!\")\n",
+    "\n",
+    "xml_generator = HelicityAmplitudeGeneratorXML()\n",
+    "xml_generator.generate(solutions)\n",
+    "xml_generator.write_to_file('model.xml')"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "Now we generate a phase space sample which is importance sampled by the\n",
+    "intensity. This can be done with the\n",
+    ":func:`.generate_importance_sampled_phsp` function. Note that you still\n",
+    "need a 'normal' phase-space sample for the :class:`.IntensityBuilderXML`."
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    ".. note::\n",
+    "    This takes around **1 min** on a Intel(R) Core(TM) i7-6820HQ CPU\n",
```

Comment:
Is this number accurate? Btw I believe the "amount/magnitude" of omega in the example is extremely low, because not many events in the omega bands are created (which means the relative amount of omega is small compared to the f)